crates/matrix-sdk/src/client/mod.rs

```diff
@@ -2026,7 +2031,8 @@ impl Client {
 
     /// The [`EventCache`] instance for this [`Client`].
     pub fn event_cache(&self) -> &EventCache {
-        &self.inner.event_cache
+        // SAFETY: always initialized in the `Client` ctor.
```

Comment:
I was worried about the potential scenario:

- a `Client` gets created, with no event cache
- a child client is created via `NotificationClient`: since it's cloning the event cache, it has `None`
- both clients enable the event cache
- now we have two event caches alive :boom: 

Hence why the event cache is always initialized in the `Client`. Are there more idiomatic Rust patterns to do deferred initialization (where it's guaranteed to happen)? `OnceCell` may not have been the best choice.

Reply:
Yeah, not sure if there's something better, unless we want to introduce locks.